src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
Yes, I know :smile:
I meant: should we just do `if (count==0) count = 1; else if (count>2) count = 2;` with error log, or rather `assert(count == 1 || count == 2)`?